### PR TITLE
Fix emulator

### PR DESCRIPTION
**Pull request by monsieurh**
# purpose
fix emulator crash since `multiple_context` merge

# status
tested on emulator
untested on hardware

# implementation
- The class `EmulatorProxy` was missing `__base_classes__`
- The wrapping function in `get_proxied_method` was trying to access `__name__` which was undefined

### CODE REVIEW

File: output/output.py [hunk 1]
```diff
@@ -49,7 +50,8 @@ def init_proxy(self, proxy):
 
     def get_proxied_method(o, proxy, method_name, sideeffect):
         method = getattr(o, method_name)
-        @wraps(method)
+
+        @wraps(method, ['__module__', '__doc__'])
```
Comment (CRImier) [inline output/output.py]: Why was this necessary, if I may ask? Just curious =)

Comment (monsieurh) [reply output/output.py]: Because the default argument was `'__module__', '__name__', '__doc__'` and `__name__` doesn't exist when wrapping emulator's function. Not quite sure why though.

Comment (CRImier) [reply output/output.py]: ...Oh, okay =D So, first thing is - you have a problem with emulator, but you're putting a workaround into the generic code that's used to work on all `OutputProxy` objects. I don't think it's a good idea architecturally, since this is hiding the symptoms instead of fixing the problem. In this case, this has led to a code that made me ask "hey, what kind of special case is this and why is it here?" - but, in general, this is making our core code less flexible in favour of leaving one non-core part unfixed.

As for fixing this - I don't have access to an emulator (still, sorry about that =( ), but I'm looking at this code and seeing this: `wraps` is wrapping the `method`, which lacks the `__name__`. We get `method` from `getattr()` on `o`, which is the `EmulatorProxy` object from `emulator.py`. I see that it defines `__getattr__`, always returning a `DummyCallableRPCObject`, which is an object that has `__call__` defined. Thus, it behaves like a function - we can do `()` on it and it'll tell the real Emulator to do something. There's one small problem, though - methods usually have `__name__` defined, when objects don't (I figured it out in REPL) - so, a `DummyCallableRPCObject` behaves like a function, but doesn't have what it takes to fully look like a function, specifically, a `__name__`, which is what `wraps` has a problem with. Funnily enough, it already has a `name` attribute defined, so I'll just rename it.

I've reverted your commit and pushed a `__name__`-based fix - it should work, but I don't have a way to test until, at least, this evening. Could you run the fix and verify that it works for you? Also, I added `emulator.py` docs - I have to admit, that file is not too readable,and I'm sorry for that - hopefully, the docs will make it a little bit more clear =)
